src/App.vue

```diff
@@ -361,37 +433,156 @@ export default {
+    homeButtonIconConfig() {
+      return {
+        enabled: true,
+        iconName: "home",
+        iconClass: this.menuButtonsIconClass,
+      };
     },
-    createButtonTextConfig() {
+    whatsNewButtonIconConfig() {
+      return {
+        enabled: true,
+        iconName: "gift",
+        iconClass: this.menuButtonsIconClass,
+      };
+    },
+    menuButtonClass() {
+      return [
+        {
+          "bg-gray-300": !this.isMenuButtonPressed,
+          "bg-primary": this.isMenuButtonPressed,
+        },
+        `rounded-lg shadow-lg ring-primary w-16 p-2`,
+      ];
+    },
+    menuButtonIconConfig() {
+      // config for the icon of menu button
+      return {
+        enabled: true,
+        iconName: "menu",
+        iconClass: this.menuIconClass,
+      };
+    },
+    menuIconClass() {
+      return [
+        {
+          "text-white": this.isMenuButtonPressed,
+          "text-black": !this.isMenuButtonPressed,
+        },
+        `fill-current h-8 w-8`,
+      ];
+    },
+    isMenuShown() {
+      return this.isMenuShownSideways || this.isMenuShownOverlay;
+    },
+    isMenuShownSideways() {
+      return this.isAuthenticated && this.isMenuButtonPressed && this.onHomePage;
+    },
+    isMenuShownOverlay() {
+      return this.isAuthenticated && this.isMenuButtonPressed && !this.onHomePage;
+    },
+    createButtonMenuTextConfig() {
       // config for the text of the main create button
       return {
         value: this.$t("home.create_button"),
-        class: "text-lg md:text-xl lg:text-2xl text-white",
+        class: "text-md sm:text-lg md:text-xl lg:text-2xl text-white",
       };
     },
-    createButtonClass() {
-      // class for the create button
-      return "bg-primary hover:bg-primary-hover rounded-lg ring-primary px-2";
+    logoutButtonTextConfig() {
+      // config for the logout button
+      return {
+        value: this.$t("nav.logout"),
+        class: this.menuButtonsTextClass,
+      };
+    },
+    whatsNewButtonTextConfig() {
+      // config for the whats new button
+      return {
+        value: this.$t("nav.whats_new"),
+        class: this.menuButtonsTextClass,
+      };
+    },
+    productGuidesButtonTextConfig() {
+      // config for the product guides button
+      return {
+        value: this.$t("nav.product_guides"),
+        class: this.menuButtonsTextClass,
+      };
+    },
+    homeButtonTextConfig() {
+      // config for the home button
+      return {
+        value: this.$t("nav.home"),
+        class: this.menuButtonsTextClass,
+      };
+    },
+    docsButtonTextConfig() {
+      // config for the plio docs button
+      return {
+        value: this.$t("nav.docs"),
+        class: this.menuButtonsTextClass,
+      };
     },
     showWorkspaceSwitcher() {
       // whether to show workspace switcher
-      return this.isAuthenticated && this.onHomePage && this.user.organizations.length;
+      return this.user.organizations.length > 0;
```

Comment:
If `isAuthenticated` is false, then `this.user` will be null and this will throw an error no? saying that cannot find property `organizations` of `null`?

Reply:
if `isAuthenticated` is false, the whole menu won't be rendered in the first place - this will only be evaluated when the menu is rendered - the `isAuthenticated` check there takes care of it and doesn't have to be repeated here.